contracts/margin/external/exchangewrappers/SimpleMarketExchangeWrapper.sol

```diff
@@ -0,0 +1,246 @@
+/*
+
+    Copyright 2018 dYdX Trading Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+pragma solidity 0.4.24;
+pragma experimental "v0.5.0";
+
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import { ISimpleMarket } from "../../../external/Maker/ISimpleMarket.sol";
+import { AdvancedTokenInteract } from "../../../lib/AdvancedTokenInteract.sol";
+import { MathHelpers } from "../../../lib/MathHelpers.sol";
+import { TokenInteract } from "../../../lib/TokenInteract.sol";
+import { ExchangeReader } from "../../interfaces/ExchangeReader.sol";
+import { ExchangeWrapper } from "../../interfaces/ExchangeWrapper.sol";
+
+
+/**
+ * @title SimpleMarketExchangeWrapper
+ * @author dYdX
+ *
+ * dYdX ExchangeWrapper to interface with Maker's (Oasis exchange) SimpleMarket or MatchingMarket
+ * contracts to trade using a specific offer. Since any MatchingMarket is also a SimpleMarket, this
+ * ExchangeWrapper can also be used for any MatchingMarket.
+ */
+contract SimpleMarketExchangeWrapper is
```

Comment:
I think the names of these contracts should have oasis in them somewhere. Otherwise it's super generally named